### ISSUE TITLE
Integrate examples from the `example-archive` into the tutorial

### ISSUE THREAD

**Issue by septract** [rems-project/cn-tutorial#39]
The `example-archive` directory has some nice examples that should be propagated back into the tutorial, with supporting text. In particular: 

- [ ] [linear search](https://github.com/rems-project/cn-tutorial/blob/main/src/example-archive/dafny-tutorial/working/linear_search.c) 
- [ ] [binary search](https://github.com/rems-project/cn-tutorial/blob/main/src/example-archive/dafny-tutorial/broken/error-crash/binary_search.c) (after fixing https://github.com/rems-project/cerberus/issues/231) 

This issue migrated from https://github.com/GaloisInc/VERSE-Toolchain/issues/31

**Comment by septract** [comment]
cc @bcpierce00

**Comment by bcpierce00** [comment]
Do we have division now?

   - B

**Comment by septract** [comment]
No, so that example is pending the issue I pointed to. I think there's a draft PR from Apol but I don't know how far they have got

**Comment by bcpierce00** [comment]
This should be unblocked now I think…

**Comment by septract** [comment]
@bcpierce00 I could take a shot at writing these up? Might be a week or two until I get time. Is the best method to generate a PR and then ask you / @cp526 for review?

**Comment by cp526** [comment]
Sounds good to me.

**Comment by bcpierce00** [comment]
Sounds great to me too!

**Comment by bcpierce00** [comment]
If you want to discuss where they belong / how best to integrate, you know
where to find us.
